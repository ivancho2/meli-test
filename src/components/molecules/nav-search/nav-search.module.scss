.nav-search {
  display: flex;
  flex: 1;
  position: relative;
  &-input {
    border-radius: var(--border-radius);
    width: 100%;
    border: none;
    font-size: 18px;
    padding: 10px var(--space-sm);
    outline: none;
    color: var(--color-black);
    padding-right: 50px;
    &::placeholder {
      color: var(--color-gray);
    }
  }
  &-btn {
    position: absolute;
    right: 0;
    top: 50%;
    transform: translateY(-50%);
    width: 45px;
    height: 100%;
    border: none;
    background-color: var(--color-gray--light);
    padding: 0;
    display: flex;
    justify-content: center;
    align-items: center;
    border-radius: 0 var(--border-radius) var(--border-radius) 0;
    outline: none;
    &:focus {
      outline: 1px dotted var(--color-blue-dodger);
    }
    &-icon {
      position: relative;
      width: 18px;
      height: 18px;
    }
  }
}
