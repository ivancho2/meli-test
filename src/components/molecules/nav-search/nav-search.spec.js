import { fireEvent, render, screen } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import { NavSearch } from './nav-search'
import * as nextRouter from 'next/router'

nextRouter.useRouter = jest.fn()
nextRouter.useRouter.mockImplementation(() => ({
  route: '/',
  query: { search: 'hello' },
}))

describe('nav-search', () => {
  it('snap', () => {
    // Arrange (Organizar/Inicializa)
    const { baseElement } = render(<NavSearch />)
    // Act (Actuar)
    // Assert (Confirmar/Comprobar)
    expect(baseElement).toMatchSnapshot('render with value')
  })

  it('set input value', () => {
    // Arrange (Organizar/Inicializa)
    const { getByPlaceholderText } = render(<NavSearch />)
    const inputSearch = getByPlaceholderText('Nunca dejes de buscar')

    // Act (Actuar)
    userEvent.clear(inputSearch)
    userEvent.type(inputSearch, 'celular')

    // Assert (Confirmar/Comprobar)
    expect(inputSearch).toHaveValue('celular')
  })

  it('sutmit event', () => {
    // Arrange (Organizar/Inicializa)
    const { getByRole } = render(<NavSearch />)
    const form = getByRole('search')
    const submitEventMock = jest.fn()

    // Act (Actuar)
    form.onsubmit = submitEventMock

    fireEvent.submit(form)
    // Assert (Confirmar/Comprobar)
    expect(submitEventMock).toBeCalledTimes(1)
  })
})
