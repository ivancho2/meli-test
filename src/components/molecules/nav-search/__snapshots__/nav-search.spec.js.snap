// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`nav-search snap: render with value 1`] = `
<body>
  <div>
    <form
      action="/items?"
      class="nav-search"
      method="GET"
      role="search"
    >
      <input
        aria-label="Ingresa lo que quieras encontrar"
        autocapitalize="off"
        autocomplete="off"
        autocorrect="off"
        class="nav-search-input"
        maxlength="120"
        name="search"
        placeholder="Nunca dejes de buscar"
        required=""
        spellcheck="false"
        tabindex="0"
        type="text"
        value="hello"
      />
      <button
        class="nav-search-btn"
        tabindex="0"
        type="submit"
      >
        <span
          class="nav-search-btn-icon"
        >
          <div
            style="display: block; overflow: hidden; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; margin: 0px;"
          >
            <noscript />
            <img
              alt="Lupa"
              decoding="async"
              src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
              style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
            />
          </div>
        </span>
      </button>
    </form>
  </div>
</body>
`;
